openedx/core/djangoapps/system_wide_roles/models.py

```diff
@@ -0,0 +1,51 @@
+"""
+Django models for system wide roles.
+"""
+from __future__ import unicode_literals
+
+from edx_rbac.models import UserRole, UserRoleAssignment
+
+
+class SystemWideRole(UserRole):  # pylint: disable=model-missing-unicode
+    """
+    User role definitions to govern non-enterprise system wide roles.
+     .. no_pii:
+    """
+
+    def __str__(self):
+        """
+        Return human-readable string representation.
+        """
+        str_representation = "<SystemWideRole {role}>"
+        return str_representation.format(role=self.name)
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
```

Comment:
Is there any particular reason you're overriding `__str__` and `__repr__`?

Reply:
Not particular reason other than that I wanted to follow the existing conventions used:
https://github.com/edx/ecommerce/blob/2b177058272c2a9b5e15b2f8f81a5de296d62c4b/ecommerce/core/models.py#L684

Reply:
@zainab-amir I believe they are doing it for a reason in ecommerce, `__str__` and `__repr__` are already defined in base class and not need to override her if there is no need https://github.com/edx/edx-rbac/blob/43c278c47bc87b48e099c71f5232af726a0b9054/edx_rbac/models.py#L96-L109

Reply:
Those are all fairly odd choices, from a python perspective. The `str` version isn't totally unreasonable, but the standard choice for `repr` is to use a string that would evaluate back to the same object.

Although, I suppose django models tend to have fairly non-standard `str` and `repr` outputs, from the perspective of other python objects.

In any event, consistency is a fine principle to stand by.

Reply:
@cpennington keeping in mind your last point ^, I have added the `str` and `repr` implementations again